### ISSUE TITLE
some questions about network partion

### ISSUE THREAD

**Issue by NoIdeaOfName66** [Vurgit/case_IEEE123#1]
Hello, I am a postgraduate student, recently learning distribution network partition and voltage control. I have seen your question in issue of DeanZeng（ https://github.com/DeanZeng/Power-gird-partition ） and learned that you have done some research on network partition. It seems that Zeng's program lacks some functions, such as 'findConnComp', may I ask if you have any solution for this? I know from the program you published （ https://github.com/Vurgit/case_IEEE123 ）that you have also read Dr. Chai's paper ‘Network partition and voltage coordination control for distribution networks with high penetration of distributed PV units’, did you find the relevant codes to replicate the network partition part? Hope to get your reply, thank you!

**Comment by Vurgit** [comment]
> Hello, I am a postgraduate student, recently learning distribution network partition and voltage control. I have seen your question in issue of DeanZeng（ https://github.com/DeanZeng/Power-gird-partition ） and learned that you have done some research on network partition. It seems that Zeng's program lacks some functions, such as 'findConnComp', may I ask if you have any solution for this? I know from the program you published （ https://github.com/Vurgit/case_IEEE123 ）that you have also read Dr. Chai's paper ‘Network partition and voltage coordination control for distribution networks with high penetration of distributed PV units’, did you find the relevant codes to replicate the network partition part? Hope to get your reply, thank you!

Not really, I did partition using other methods as I described in paper "Y. Yang, Y. Sun, Q. Wang, F. Liu and L. Zhu, "Fast Power Grid Partition for Voltage Control With Balanced-Depth-Based Community Detection Algorithm," in IEEE Transactions on Power Systems, vol. 37, no. 2, pp. 1612-1622, March 2022, doi: 10.1109/TPWRS.2021.3107847."

**Comment by Vurgit** [comment]
> Hello, I am a postgraduate student, recently learning distribution network partition and voltage control. I have seen your question in issue of DeanZeng（ https://github.com/DeanZeng/Power-gird-partition ） and learned that you have done some research on network partition. It seems that Zeng's program lacks some functions, such as 'findConnComp', may I ask if you have any solution for this? I know from the program you published （ https://github.com/Vurgit/case_IEEE123 ）that you have also read Dr. Chai's paper ‘Network partition and voltage coordination control for distribution networks with high penetration of distributed PV units’, did you find the relevant codes to replicate the network partition part? Hope to get your reply, thank you!

Not really, I did partition using other methods as I described in paper "Y. Yang, Y. Sun, Q. Wang, F. Liu and L. Zhu, "Fast Power Grid Partition for Voltage Control With Balanced-Depth-Based Community Detection Algorithm," in IEEE Transactions on Power Systems, vol. 37, no. 2, pp. 1612-1622, March 2022, doi: 10.1109/TPWRS.2021.3107847."

**Comment by orangecatword** [comment]
I was recently reading your article “Fast Power Grid Partition for Voltage Control With Balanced-Depth-Based Community Detection Algorithm,” could you please share the program in it?

**Comment by LpHouse** [comment]
> I was recently reading your article “Fast Power Grid Partition for Voltage Control With Balanced-Depth-Based Community Detection Algorithm,” could you please share the program in it?

The codes connot be made available right now because it has some tiny problems that I don't want to show :(; however, I can tell you how can you replicate the results: You can download python package igraph to do network partition, and the sensitivity matrix of the power network can be easily obtained using matpower functions dSbus_dVm and dSbus_Va; the voltage control part can be obtained using the methods described in paper "Discovering Clusters in Power Networks From Orthogonal Structure of Spectral Embedding", where he shared his code generously. The selection of pilot buses can be implemented using matlab function surrogateopt(). I hope this helps.

**Comment by orangecatword** [comment]
> 代码现在不能公开，因为它有一些小问题，我不想让:(看到;不过，我可以告诉你如何复现结果：你可以下载python包igraph来做网络划分，电力网络的灵敏度矩阵可以通过matpower函数dSbus_dVm和dSbus_Va轻松获得;电压控制部分可以通过他在论文《从谱嵌入的正交结构发现电力网络中的簇》中描述的方法获得，他在文中慷慨分享了自己的代码。试点总线的选择可以通过 matlab 函数 surrogateopt（） 实现。希望这些对你有帮助。

Thank you for your help.<img width="48" height="48" alt="Image" src="https://github.com/user-attachments/assets/5e6bf113-45be-4399-9112-30e8194776ec" />

**Comment by LpHouse** [comment]
There are also some relevant codes that can be found on github, such as 
[https://github.com/NoahCherry13/Distbution_Network_Optimization/tree/0a7407366fcd28bc4e9325c6fd28ba869fef8ce1](url)
[https://github.com/SumitBahl02/Voltage_Control_Areas](url)
They may be better than my implementation.

**Comment by orangecatword** [comment]
> There are also some relevant codes that can be found on github, such as [https://github.com/NoahCherry13/Distbution_Network_Optimization/tree/0a7407366fcd28bc4e9325c6fd28ba869fef8ce1](url) [https://github.com/SumitBahl02/Voltage_Control_Areas](url) They may be better than my implementation.

Your help is much appreciated,Thanks!!!!!😊
